### PR TITLE
Rust: Sync Cargo.toml project metadata attributes

### PR DESCRIPTION
**Pull request by jayvdb**
Sync Cargo.toml project metadata attributes.  The client and server Cargo templates both need enhancements to be similar and allow full control over the project metadata. 

ping @frol @farcaller @richardwhiuk @paladinzh @jacob-pro

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.1.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by jacob-pro** [comment]
Looks good to me :)
@wing328 should be able to review

File: modules/openapi-generator/src/main/resources/rust/Cargo.mustache [hunk 1]
```diff
@@ -1,8 +1,26 @@
 [package]
 name = "{{{packageName}}}"
 version = "{{#lambdaVersion}}{{{packageVersion}}}{{/lambdaVersion}}"
-authors = ["OpenAPI Generator team and contributors"]
+authors = [{{#infoEmail}}"{{{.}}}"{{/infoEmail}}]
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/rust/Cargo.mustache:4]: can we add back "OpenAPI Generator team and contributors" if infoEmail is not provided in the spec?

File: modules/openapi-generator/src/main/resources/rust-server/Cargo.mustache [hunk 2]
```diff
@@ -5,7 +5,9 @@ authors = [{{#infoEmail}}"{{{.}}}"{{/infoEmail}}]
 {{#appDescription}}
 description = "{{{.}}}"
 {{/appDescription}}
-license = "Unlicense"
+{{#licenseInfo}}
+license = "{{.}}"
+{{/licenseInfo}}
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/rust-server/Cargo.mustache:10]: if licenseInfo is not provided, can we default it to `license = "Unlicense ` ?

Comment (jayvdb) [reply modules/openapi-generator/src/main/resources/rust-server/Cargo.mustache:10]: Sure.  Can I also add a comment explaining that the license can be set using the `info.license` field in the openapi spec?

Comment (wing328) [reply modules/openapi-generator/src/main/resources/rust-server/Cargo.mustache:10]: Im ok with that.

**Pull request merged**